File: src/main/kotlin/codeacademy/bookingforum/app/report/ReportDto.java
package codeacademy.bookingforum.app.report;

import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ReportDto {
    Long id;
    @NotEmpty(message = "Please specify a reason for this report.")
    String reason;
    @NotEmpty(message = "Please describe the issue.")
    String description;
    LocalDateTime reportDate;
    Long userId;
    Long postId;
    Long commentId;
}

File: src/main/kotlin/codeacademy/bookingforum/app/report/ReportDto.kt
package codeacademy.bookingforum.app.report

import java.time.LocalDateTime

data class ReportDto(
    val id: Long? = null,

    val reason: String? = "",

    val description: String? = "",

    val reportDate: LocalDateTime? = null,

    val commentId: Long? = null,

    val userID: Long? = null,

    val postId: Long? = null,
)
